### DISCUSSION TITLE
Query About Handling OR Conditions in Parent-Child Relationships in the OCA Package

### DISCUSSION THREAD

**Discussion by setayesh78** [the-human-colossus-foundation/oca-spec#75] [General]
I have a question about the functionality of the OCA package regarding parent-child relationships. Specifically, we might have a requirement where a parent references two child items, and the user can select between them using an OR condition. For example:
The user selects an option via a radio button and gets redirected to one child reference.
Alternatively, the user selects the other option and gets redirected to the other child reference.
Does the OCA package currently support this functionality, or is it planned for implementation in the future?
Thanks in advance for clarifying!

**Comment by blelump** [comment]
OCA mimics the RDBMS behavior when thinking of relationships. We discuss them in a [relationships](https://oca.colossi.network/ecosystem/ocafile.html#relationships) section (assuming you're using OCAFiles). Modeling a case you're referring to seems to require
```
--name=User
ADD ATTRIBUTE child1=refn:Child1 child2=refn:Child2
ADD CONFORMANCE ATTRS child1=O child2=O
```
which means both relations are optional. OCA doesn't support `one of` on the semantics level. We leave it to the upper levels to ensure either A or B is present.

It is also worth noting the behavior of how it ends up on the screen, i.e., as a WEB form where, depending on the radio value, `child1` or `child2` attributes pop up on the screen. This is a case-specific problem, specifically to handle such cases as `IF value matches X then show Y`.

**Comment by mitfik** [comment]
Worth to add is that there is already a project which deals with presentation and glue semantic with presentation all together, it is used in various use cases like digital forms, you can read more about it here: https://github.com/THCLab/oca-interpres
specifically [tests](https://github.com/THCLab/oca-interpres/blob/main/src/index.test.ts) would give you an idea how to leverage it.  

The conditional aspect is application layer problem (business logic) so needs to be implemented separately.
